flang/include/flang/Runtime/CUDA/memory.h

```diff
@@ -17,12 +17,24 @@ namespace Fortran::runtime::cuda {
 
 extern "C" {
 
+/// Allocate memory on the device.
+void *RTDECL(CUFMemAlloc)(
```

Comment:
Do you plan to have different APIs for different CUF attributes or should we pass the device/managed/etc. attribute to the APIs to use different allocation/free methods?

Reply:
This is initially for device only. For managed/unified right now they are only supported through allocators. I guess when we will have broader support we can either have separate entry points of pass the information to these functions.